lms/djangoapps/email_marketing/tests/test_signals.py

```diff
@@ -0,0 +1,169 @@
+"""Tests of email marketing signal handlers."""
+import logging
+
+from django.test import TestCase
+from django.test.utils import override_settings
+from mock import patch
+from util.json_request import JsonResponse
+
+from email_marketing.signals import handle_unenroll_done, \
+    email_marketing_register_user, \
+    email_marketing_user_field_changed
+from email_marketing.tasks import update_user, update_user_email
+from email_marketing.models import EmailMarketingConfiguration
+from django.test.client import RequestFactory
+from student.tests.factories import UserFactory, UserProfileFactory
+
+from sailthru.sailthru_client import SailthruClient
+from sailthru.sailthru_response import SailthruResponse
+from sailthru.sailthru_error import SailthruClientError
+
+log = logging.getLogger(__name__)
+
+
+def update_email_marketing_config(enabled=False, key='badkey', secret='badsecret', new_user_list='new list',
+                                  template='Activation'):
+    """
+    Enable / Disable Sailthru integration
+    """
+    EmailMarketingConfiguration.objects.create(
+        enabled=enabled,
+        sailthru_key=key,
+        sailthru_secret=secret,
+        sailthru_new_user_list=new_user_list,
+        sailthru_activation_template=template
+    )
+
+
+class EmailMarketingTests(TestCase):
+    """
+    Tests for the EmailMarketing signals and tasks classes.
+    """
+
+    def setUp(self):
+        self.request_factory = RequestFactory()
+        self.user = UserFactory.build(username='test', email='test@edx.org')
+        self.profile = UserProfileFactory.build(user=self.user)
```

Comment:
Because of [this method](https://github.com/edx/edx-platform/blob/d53e4f2ed26c717e8d27ffd099e8d165c555c0e6/common/djangoapps/student/tests/factories.py#L95-L103), you don't have to create the profile manually, you could just use `self.profile = user.profile`.